google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -618,25 +620,13 @@ protected RetrySettings defaultRetrySettings() {
     return Iterables.getFirst(ServiceLoader.load(clazz), defaultInstance);
   }
 
-  private static String defaultLibraryVersion() {
-    String version = getPomVersion();
-    if (version == null) {
-      version = getManifestVersion();
-    }
-    return version;
-  }
-
-  private static String getPomVersion() {
+  private static String getPomVersion(String artifactId) {
     try {
-      Properties properties = new Properties();
-      String mavenPropertiesPath = META_FILE_ROOT
-          + ServiceOptions.class.getPackage().getName() + "/"
-          + ARTIFACT_ID + "/pom.properties";
-      InputStream inputStream = ServiceOptions.class.getResourceAsStream(mavenPropertiesPath);
-      if (inputStream != null) {
-        properties.load(inputStream);
-        return properties.getProperty(META_VERSION_KEY, "");
-      }
+      String projectPropertiesPath = PROPERTIES_ROOT
+          + artifactId + "/"
+          + PROPERTIES_FILE;
```

Comment:
correctness verification would be easier if `PROPERTIES_ROOT` didn't have the final slash, and the slash was put here: someone could easily see by eye that all of the necessary slashes were present.